moabb/datasets/castillos2023.py

```diff
@@ -0,0 +1,542 @@
+        # onset_window = np.arange(0,time_trial*fps*(nb_seq_max-nb_seq_min)-1,1,dtype=int)
+        for i, o in enumerate(onset_window):
+            if label_window[i] == 1:
+                # print(i)
+                if current_code == nb_seq_max - 1 - nb_seq_min:
+                    new_onset.append(o + onset_shift)
+                else:
+                    if (
+                        o + onset_shift
+                        >= onset_code[current_code + nb_seq_min] + time_trial * self.fps
+                    ):
+                        current_code += 1
+                        onset_shift = (
+                            onset_code[current_code + nb_seq_min]
+                            - time_trial * self.fps * current_code
+                        )
+                    new_onset.append(o + onset_shift)
+            else:
+                if current_code == nb_seq_max - 1 - nb_seq_min:
+                    new_onset_0.append(o + onset_shift)
+                else:
+                    if (
+                        o + onset_shift
+                        >= onset_code[current_code + nb_seq_min] + time_trial * self.fps
+                    ):
+                        current_code += 1
+                        onset_shift = (
+                            onset_code[current_code + nb_seq_min]
+                            - time_trial * self.fps * current_code
+                        )
+                    new_onset_0.append(o + onset_shift)
+
+        # modified_onset_code = [onset_code[i]-time_trial*sfreq*i for i in range(nb_seq_min,nb_seq_max)]
+        # new_onset_0 = np.concatenate([np.arange(onset_code[i],onset_code[i]+time_trial*sfreq,sfreq//60) for i in range(nb_seq_min,nb_seq_max)])
+        new_onset_0 = np.array(list(filter(lambda i: i not in new_onset, new_onset_0)))
+        # print(new_onset_0.shape)
+        return np.array(new_onset) / self.fps, np.array(new_onset_0) / self.fps
+
+
+class CasitllosBurstVEP100(BaseCastillos2023):
+    """BurstVEP CasitllosBurstVEP100 dataset.
+
+    .. admonition:: Dataset summary
+
+
+        ====================      =======  =======  ==========  =================  ===============  ===============  ===========
+        Name                      #Subj    #Chan    #Classes    #Trials / class    Trials length    Sampling rate      #Sessions
+        ====================      =======  =======  ==========  =================  ===============  ===============  ===========
+        CasitllosBurstVEP100       12      32            4      15                 2.2s               500Hz                    1
+        ====================      =======  =======  ==========  =================  ===============  ===============  ===========
+
+    """
+
+    def __init__(self):
+        super().__init__(
+            events={"0": 100, "1": 101},
+            sessions_per_subject=1,
+            code="CasitllosBurstVEP100",
+            paradigm="cvep",
+            paradigm_type="burst100",
```

Comment:
I think I understand what you want to do with the `paradigm_type` arg. I think you should remove it for now. You could properly add it the CVEP class once this PR is merged.